src/app/admin/subject/[slug]/page.tsx

```diff
@@ -0,0 +1,185 @@
+  "AP German",
+  "AP Human Geography",
+  "AP Italian",
+  "AP Japanese",
+  "AP Latin",
+  "AP Macroeconomics",
+  "AP Microeconomics",
+  "AP Music Theory",
+  "AP Physics 1",
+  "AP Physics 2",
+  "AP Physics C: E&M",
+  "AP Physics C: Mechanics",
+  "AP Precalculus",
+  "AP Psychology",
+  "AP Research",
+  "AP Seminar",
+  "AP Spanish Language",
+  "AP Spanish Literature",
+  "AP Statistics",
+  "AP US History",
+  "AP United States Government",
+  "AP World History: Modern",
+].sort();
+
+// set empty data if non-existent
+const emptyData: Subject = {
+  title: "",
+  units: [
+    {
+      unit: 1,
+      title: "",
+      chapters: [
+        {
+          chapter: 1,
+          title: "",
+        },
+      ],
+    },
+  ],
+};
+
+const Page = ({ params }: { params: { slug: string } }) => {
+  const [subject, setSubject] = useState<Subject | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        const fetchedUser = await getUser();
+        setUser(fetchedUser);
+      } catch (error) {
+        console.error("Error fetching user:", error);
+      }
+    };
+
+    fetchUser();
+  }, []);
+
+  useEffect(() => {
+    const fetchSubject = async () => {
+      try {
+        if (user && (user?.access === "admin" || user?.access === "member")) {
+          // Reference to the document in Firestore using the slug
+          const docRef = doc(db, "subjects", params.slug);
+          const docSnap = await getDoc(docRef);
+          if (docSnap.exists()) {
+            // Convert Firestore document data to Subject type
+            setSubject(docSnap.data() as Subject);
+          } else {
+            emptyData.title = apClasses.find((apClass) => apClass.replace(/AP /g, "").toLowerCase().replace(/[^a-z1-9 ]+/g, "").replace(/\s/g, "-") === params.slug) || "";
+            setSubject(emptyData);
+          }
+        }
+      } catch (error) {
+        console.log("Error fetching subject data:", error);
+        setError("Failed to fetch subject data.");
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchSubject();
+  }, [user]);
+
+  if (loading) {
+    return (
+      <div className="flex min-h-screen items-center justify-center text-3xl">
+        Loading...
+      </div>
+    );
+  }
+
+  if (error || !subject) {
+    return (
+      <div className="flex min-h-screen items-center justify-center text-3xl">
+        {error}
+      </div>
+    );
+  }
+
+  const handleSave = async () => {
+    try {
+      if (user && (user?.access === "admin" || user?.access === "member")) {
+
+        // Save to Firestore
+        await setDoc(doc(db, "subjects", params.slug), subject);
+
+        setSubject(subject);
+      } else {
+        console.error("Error saving new unit and chapter:", error);
+      }
```

Comment:
I feel like this isn't correct

Reply:
i agree, in the case that the user doesn't have the perms to save a document, you want to give them a clear error, not something in the console which they aren't gonna see
maybe an alert or something would be simplest

Reply:
nvm i see this is an admin gated page that's mb